pinot-query-planner/src/main/java/org/apache/calcite/rel/hint/PinotHintStrategyTable.java

```diff
@@ -16,19 +16,21 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.pinot.query.planner.hints;
-
-import org.apache.calcite.rel.hint.RelHint;
-
+package org.apache.calcite.rel.hint;
 
 /**
- * Provide certain relational hint to query planner for better optimization.
+ * Default hint strategy set for Pinot query.
  */
-public class PinotRelationalHints {
-  public static final RelHint AGG_INTERMEDIATE_STAGE = RelHint.builder("AGG_INTERMEDIATE_STAGE").build();
-  public static final RelHint AGG_LEAF_STAGE = RelHint.builder("AGG_LEAF_STAGE").build();
+public class PinotHintStrategyTable {
+  public static final String INTERNAL_AGG_INTERMEDIATE_STAGE = "aggIntermediateStage";
```

Comment:
Basic question - if the user passes a hint that contains both or one of the `aggIntermediateStage` and `aggFinalStage` hints, wouldn't we completely skip applying the PinotAggregateExchangeNodeInsertRule RelOptRule because of the `PinotAggregateExchangeNodeInsertRule.matches` function will evaluate to false? Is this the expected behavior?

Reply:
yes. again whether a hint is valid or not is up to the implementation. but that's actually a good point. i can make this as `aggregateHint(final)` and `aggregateHint(intermediate)` this way we only have 1 hint key but different hint options. 

i think there's a restriction on whether a hint can accept multiple option key in order to make this more fool proof